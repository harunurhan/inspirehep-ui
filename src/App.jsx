import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import { Layout, Menu, Row, Col } from 'antd';

import './App.css';
import Holdingpen from './holdingpen';
import Literature from './literature';
import SearchBoxContainer from './common/containers/SearchBoxContainer';


class App extends Component {
  render() {
    return (
      <div>
        <Layout>
          <Layout.Header>
            <Row>
              <Col span={12}>
                <Menu
                  className="menu"
                  theme="dark"
                  mode="horizontal"
                >
                  <Menu.Item>
                    <Link to="/holdingpen/dashboard">Holdingpen</Link>
                  </Menu.Item>
                </Menu>
              </Col>
              <Col span={12}>
                <SearchBoxContainer />
              </Col>
            </Row>
          </Layout.Header>

          <Layout.Content>
            <Route path="/holdingpen" component={Holdingpen} />
            <Route path="/literature" component={Literature} />
          </Layout.Content>
        </Layout>
      </div>
    );
  }
}

export default App;
