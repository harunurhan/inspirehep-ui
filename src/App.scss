.__App__ {
  min-height: 100vh;
  .content {
    padding: 0 50px;
    margin-top: 64px;
  }
}
