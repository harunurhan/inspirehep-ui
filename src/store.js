import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
