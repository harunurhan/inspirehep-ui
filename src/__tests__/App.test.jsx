import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import { getStore } from '../fixtures/store';
import App from '../App';
import Holdingpen from '../holdingpen';
import Home from '../home';
import Literature from '../literature';

describe('App', () => {
  it('renders initial state', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it('navigates to Holdingpen when /holdingpen', () => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={['/holdingpen']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(Holdingpen)).toExist();
  });

  it('navigates to Literature when /literature', () => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={['/literature']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(Literature)).toExist();
  });

  it('navigates to Home when /', () => {
    const wrapper = mount(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(Home)).toExist();
  });
});
