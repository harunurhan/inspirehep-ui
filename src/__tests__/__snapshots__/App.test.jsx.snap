// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders initial state 1`] = `
<div>
  <Adapter>
    <Adapter>
      <Row
        gutter={0}
      >
        <Col
          span={12}
        >
          <Menu
            className="menu"
            mode="horizontal"
            prefixCls="ant-menu"
            theme="dark"
          >
            <MenuItem>
              <Link
                replace={false}
                to="/holdingpen/dashboard"
              >
                Holdingpen
              </Link>
            </MenuItem>
          </Menu>
        </Col>
        <Col
          span={12}
        >
          <Connect(SearchBox) />
        </Col>
      </Row>
    </Adapter>
    <Adapter>
      <Route
        component={[Function]}
        path="/holdingpen"
      />
      <Route
        component={[Function]}
        path="/literature"
      />
    </Adapter>
  </Adapter>
</div>
`;
